seisbench/generate/augmentation.py

```diff
@@ -3,6 +3,8 @@
 import numpy as np
 import scipy.signal
 
+import seisbench.data.base
```

Comment:
So far, the idea was that the three main modules of SeisBench do not import each other (https://github.com/seisbench/seisbench/blob/main/CONTRIBUTING.md#code-structure). From what I see, you only need it for the type hints here. I'd suggest removing the type hints for the WaveformDataset to get rid of the import. But maybe we should consider changing the policy and allow to import seisbench.data in seisbench.generate?

Reply:
I fully understand that point, especially to avoid circular imports.
However, at the moment, I do not have an idea how to avoid this import, since it is needed for the type hint and if I don't import the module, I get an error in line 811 for the noise generator. If you have an idea, let me know.

Reply:
I was thinking of just dropping the type hint here altogether, but that's admittedly not a great solution. I'll update the contributing document.